lib/generator.dart

```diff
@@ -17,12 +18,30 @@ LibraryDefinition generateLibrary(
   GeneratorOptions options,
   SchemaMap schemaMap,
 ) {
+  // TODO this class deduplication doesn't work (had to use prefixes),
+  //      because different queries have on different selection sets
+  // To me this is more evidence that generating full restricted types that are then
```

Comment:
This approach is even more appealing now with [static extensions](https://github.com/dart-lang/language/blob/master/accepted/2.6/static-extension-members/feature-specification.md#syntax)

Reply:
don't think static extensions will because of multiple selection sets in the same query

Reply:
What's your idea on "full restricted types loosened"? I agree that approach would make the code cleaner, and I'm up to it, but I can't see how it'd solve types without prefixes.